metadata-models/src/main/pegasus/com/linkedin/common/Cost.pdl

```diff
@@ -0,0 +1,12 @@
+namespace com.linkedin.common
+
+/*
+* Cost Details for an Entity
+*/
+record Cost {
```

Comment:
Within Expedia we have a costCode attached to at the component/project level. So based on the component/project the model belongs to this code can be attached to the data.

Do you have any other ideas to improve the Cost aspect and make it generic?

Reply:
One option might be to define an enum for type of Cost called something like CostType. It could be an enum to identify the different type of costs that can be attached. To start with, we can have ORG_COST_CODE as one of the types. Then, we can define cost value as a object which can be a union of primitive/complex cost objects types. Something similar to a model like below:

```
/*
* Cost Details for an Entity
*/
record Cost {

    costType: CostType

    cost: CostValue
}
```

```
enum CostType { 

  /** 
   * Org Cost Code to which the Cost of this entity should be attributed to
   */ 
  ORG_COST_CODE 

// additional cost types here
} 
```

```
typeref CostValue = union[ 
  numericalCost: double
  stringCost: string
  customCost: customCostModel
  ....
]
``` 

Referencing the above model, we can now have an array of Cost aspect associated with the Model entity. The above model will provide an option for others to expand it to include their own cost definitions.

I'd encourage for us to think of all common models in this way so that they are extensible to other use cases. Let me know what you think.